### PR TITLE
Fix read authorization_metadata_key from public_client_config in _refresh_credentials_from_command

### PR DESCRIPTION
**Pull request by sonjaer**
# TL;DR

`FLYTE_CREDENTIALS_AUTHORIZATION_METADATA_KEY` environment variable was removed in this PR https://github.com/flyteorg/flytekit/pull/858 in favor of reading the authorization metadata key from Admin. 

There was a miss in the PR where the `_refresh_credentials_from_command` did not switch to get the authorization key from self.public_client_config.authorization_metadata_key but instead used the default value `authorization`. 

Spotify uses `flyte-authorization` so we received an error about calls being UNAUTHENTICATED since the header name didn't correspond to the one expected by admin. 

This PR adds support for reading `authorization_metadata_key` from `public_client_config` in `_refresh_credentials_from_command`

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: flytekit/clis/flyte_cli/main.py [hunk 1]
```diff
@@ -277,6 +277,7 @@ def _get_client(host: str, insecure: bool) -> _friendly_client.SynchronousFlyteC
     if parent_ctx.obj["cacert"]:
         kwargs["root_certificates"] = parent_ctx.obj["cacert"]
     cfg = parent_ctx.obj["config"]
+    print(cfg)
```
Comment (sonjaer) [inline flytekit/clis/flyte_cli/main.py:280]:
```suggestion
```

File: flytekit/configuration/internal.py [hunk 2]
```diff
@@ -65,6 +65,17 @@ class Credentials(object):
     This command is executed to return a token using an external process.
     """
 
+    AUTHORIZATION_METADATA_KEY = ConfigEntry(LegacyConfigEntry(SECTION, "authorization_metadata_key", list), YamlConfigEntry("admin.authorizationHeader", list))
```
Comment (sonjaer) [inline flytekit/configuration/internal.py:68]: I think we could remove this actually

File: flytekit/clients/raw.py [hunk 3]
```diff
@@ -264,7 +265,7 @@ def _refresh_credentials_from_command(self):
         :return:
         """
 
-        command = self._cfg.command
+        command = self._cfg.command or Credentials.COMMAND.read()
```
Comment (eapolinario) [inline flytekit/clients/raw.py:268]: We shouldn't need this after https://github.com/flyteorg/flytekit/pull/1063/. 

If you follow click's code, `self._cfg` in the case of `flyte-cli` commands is set [here](https://github.com/flyteorg/flytekit/blob/master/flytekit/clis/flyte_cli/main.py#L514). Notice the use of `PlatformConfig.auto()`, this should [take care](https://sourcegraph.com/github.com/flyteorg/flytekit@87d13903993d5339723ecd859bd362a64e780aec/-/blob/flytekit/configuration/__init__.py?L333-346) of loading the config values using env vars if those are set.

File: tests/flytekit/unit/clients/test_raw.py [hunk 4]
```diff
@@ -195,6 +225,15 @@ def test_basic_strings(mocked_method):
...
-    cc = RawSynchronousFlyteClient(PlatformConfig(auth_mode=AuthType.EXTERNAL_PROCESS))
+    cc = RawSynchronousFlyteClient(PlatformConfig(auth_mode=AuthType.EXTERNALCOMMAND))
+    cc.refresh_credentials()
+    assert mocked_method.called
+
+
+@patch("flytekit.configuration.internal.Credentials.AUTH_MODE.read")
+@patch.object(RawSynchronousFlyteClient, "_refresh_credentials_from_command")
+def test_refresh_from_environment_variable(mocked_method, mock_read_auth_mode_from_env):
+    cc = RawSynchronousFlyteClient(PlatformConfig(auth_mode=None))
```
Comment (eapolinario) [inline tests/flytekit/unit/clients/test_raw.py:236]: Can you use `PlatformConfig.auto` and mock the relevant env vars using https://docs.pytest.org/en/6.2.x/reference.html#pytest.MonkeyPatch.setenv ?

**Review by eapolinario — COMMENTED** [review]
Thanks for this fix, the change to pass the authorization key in the case of external commands was missing.

File: flytekit/clients/raw.py (hunk 3 above)
Comment (sonjaer) [reply flytekit/clients/raw.py:268]: Ah ok sorry, missed that

**Comment by sonjaer** [comment]
@eapolinario  what takes precedence from config, env and passing a flag?

**Comment by eapolinario** [comment]
> @eapolinario what takes precedence from config, env and passing a flag?

`flyte-cli` rules are a bit more complicated, but in general, for configuration settings, environment variables take precedence over config data, with the exception of the `host` and `insecure` flags.

**Review by eapolinario — APPROVED** [review]
